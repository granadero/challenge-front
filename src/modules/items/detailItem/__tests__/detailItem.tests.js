import React from "react";
import { render } from "@testing-library/react";
import DetailItem from "../detailItem";

describe("DetailItem component", () => {
  const item = {
    picture: { url: "http://example.com/image.jpg" },
    title: "Item Test",
    condition: "Nuevo",
    sold_quantity: 10,
    price: { amount: 100, currency: "ARS", decimals: 0 },
    description: "This is an example item",
  };

  it("renders with the correct structure", () => {
    const { container } = render(<DetailItem item={item} />);
    expect(container.querySelector("section")).toBeInTheDocument();
    expect(container.querySelector("div.container")).toBeInTheDocument();
    expect(container.querySelector("div.imageAndPrice")).toBeInTheDocument();
    expect(container.querySelector("div.priceAndTitle")).toBeInTheDocument();
    expect(container.querySelector("div.description")).toBeInTheDocument();
  });

  it("does not render decimals when they are 0", () => {
    const { container } = render(<DetailItem item={item} />);
    const decimals = container.querySelector("h1.decimals");
    expect(decimals).not.toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(<DetailItem item={item} />);
    expect(container).toMatchSnapshot();
  });
});
