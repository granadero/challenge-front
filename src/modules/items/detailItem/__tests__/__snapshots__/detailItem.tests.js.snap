// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailItem component should match snapshot 1`] = `
<div>
  <section
    class="section"
  >
    <div
      class="container"
    >
      <div
        class="imageAndPrice"
      >
        <img
          alt="Item Test"
          data-nimg="1"
          decoding="async"
          height="500"
          loading="lazy"
          src="/_next/image?url=http%3A%2F%2Fexample.com%2Fimage.jpg&w=750&q=75"
          srcset="/_next/image?url=http%3A%2F%2Fexample.com%2Fimage.jpg&w=384&q=75 1x, /_next/image?url=http%3A%2F%2Fexample.com%2Fimage.jpg&w=750&q=75 2x"
          style="color: transparent;"
          width="360"
        />
        <div
          class="priceAndTitle"
        >
          <div
            class="conditionAndSolds"
          >
            Nuevo
             | 
            10
             vendidos
          </div>
          <h1
            class="title"
          >
            Item Test
          </h1>
          <div
            class="amount"
          >
            <h1
              class="price"
            >
              $ 100
            </h1>
          </div>
          <button
            class="button"
          >
            Comprar
          </button>
        </div>
      </div>
      <hr
        color="#d2d2d2"
      />
      <div
        class="description"
      >
        <h2
          class="titleDescription"
        >
          Descripción del producto
        </h2>
        <p
          class="descriptionText"
        >
          This is an example item
        </p>
      </div>
    </div>
  </section>
</div>
`;
